sound/soc/intel/boards/sof_ssp_amp.c

```diff
@@ -75,126 +75,64 @@ static struct snd_soc_card sof_ssp_amp_card = {
 #define HDMI_IN_BE_ID		0
 #define SPK_BE_ID		2
 #define DMIC01_BE_ID		3
-#define DMIC16K_BE_ID		4
 #define INTEL_HDMI_BE_ID	5
-
-static struct snd_soc_dai_link *
-sof_card_dai_links_create(struct device *dev, enum sof_ssp_codec amp_type,
-			  int ssp_amp, int dmic_be_num, int hdmi_num,
-			  bool idisp_codec)
+/* extra BE links to support no-hdmi-in boards */
+#define DMIC16K_BE_ID		4
+#define BT_OFFLOAD_BE_ID	8
+
+#define NO_CODEC_LINK_ORDER	SOF_LINK_ORDER(SOF_LINK_HDMI_IN, \
+					SOF_LINK_AMP,            \
+					SOF_LINK_DMIC01,         \
+					SOF_LINK_DMIC16K,        \
+					SOF_LINK_IDISP_HDMI,     \
+					SOF_LINK_BT_OFFLOAD,     \
+					SOF_LINK_NONE)
+
+#define NO_CODEC_LINK_IDS	SOF_LINK_ORDER(HDMI_IN_BE_ID, \
```

Comment:
Sorry, I am not following what this 'NO_CODEC' means?

We already have a nocodec mode, but then we don't use this machine driver at all.

I guess I must be missing a lot of context on what you are trying to do @brentlu

Reply:
It just means the boards without headphone codec. Sorry for the confusing with the nocodec mode. I will rename the macros.